### PR TITLE
[Move] Implement Snatch

### PR DESCRIPTION
**Pull request by innerthunder**
## What are the changes the user will see?

Snatch is now fully implemented.

## Why am I making these changes?

📈 

## What are the changes from a developer perspective?

- Added the `SNATCHABLE` move flag for moves that can have their effects stolen by Snatch.
- Added a `SnatchTag` battler tag to indicate a Pokemon is able to steal a snatchable move's effects and queue messages during the move's execution.
- Added logic to `MovePhase` similar to Magic Coat/Bounce to handle when a move is stolen by another Pokemon's Snatch.

**Note**: Since Snatch isn't available in mainline after Gen 7, some new moves that match the characteristics of snatchable moves have been given the `SNATCHABLE` flag:
- Stuff Cheeks
- No Retreat
- Clangorous Soul
- Life Dew
- Jungle Healing
- Power Shift
- Victory Dance
- Shelter
- Lunar Blessing
- Take Heart
- Fillet Away
- Shed Tail
- Tidy Up
- Revival Blessing

## How to test the changes?

`npm run test snatch`
`npm run test all-moves` (now verifies the `SNATCHABLE` flag)

## Checklist

- [ ] ⚠️ If this is a PR for `main` (such as a hotfix), has the game version been updated (`npm run update-version:patch` / `npm run update version:minor`?
- [x] Otherwise: **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes manually?

- [x] Are all unit tests still passing? (`npm run test:silent`)
  - [x] Have I created new automated tests (`npm run test:create`) or updated existing tests related to the PR's changes?
- [ ] Have I provided screenshots/videos of the changes (if applicable)?
  - [ ] Have I made sure that any UI change works for both UI themes (dark and light)?

#### Are there any localization additions or changes? If so:

- [x] Has a locales PR been created on the [locales](https://github.com/despair-games/poketernity-locales) repo?
  - [x] If so, please leave a link to it here: https://github.com/Despair-Games/poketernity-locales/pull/41
- [x] Have I added the `Localization` tag to this PR?

### CODE REVIEW

File: src/data/init/init-moves.ts [hunk 1]
```diff
@@ -2705,13 +2780,15 @@ export function initMoves() {
     new SelfStatusMove(MoveId.STUFF_CHEEKS, ElementalType.NORMAL, -1, 10, -1, 0, 8)
       .attr(EatBerryAttr, true)
       .attr(StatStageChangeAttr, [Stat.DEF], 2, true)
+      .snatchable()
```
Comment (Tempo-anon) [inline src/data/init/init-moves.ts]: I think it'd be good to leave a comment for all the moves that don't officially have the snatchable aspect but are custom given it because they make sense.

File: test/data/all-moves.json [hunk 2]
```diff
@@ -12732,7 +12732,7 @@
     "damage_class_id": 1,
     "effect_id": "",
     "effect_chance": -1,
-    "flags": ""
+    "flags": [6, 13]
```
Comment (Tempo-anon) [inline test/data/all-moves.json:12735]: ~~I assume the 6 flag is for snatch, but what's the 13 flag?~~ Oh the triage flag was forgotten

File: test/moves/snatch.test.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    expect(playerPokemon[0].getStatStage(Stat.ATK)).toBe(2);
+    expect(enemyPokemon[0].getStatStage(Stat.ATK)).toBe(0);
+    expect(enemyPokemon[1].getStatStage(Stat.ATK)).toBe(2);
+  });
+
+  it.each([
+    { moveId: MoveId.REST, moveName: "Rest" },
+    { moveId: MoveId.SWALLOW, moveName: "Swallow" },
+  ])("should not steal $moveName when it has no effect on the original user", async ({ moveId }) => {
```
Comment (Tempo-anon) [inline test/moves/snatch.test.ts:75]: Should probably also includes a happy case for these

File: src/data/battler-tags/snatch-tag.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import i18next from "i18next";
+
+/**
+ * Tag to indicate the owner can steal a beneficial self-status move's effects
+ * with {@link https://bulbapedia.bulbagarden.net/wiki/Snatch_(move) | Snatch}.
+ * @extends BattlerTag
+ */
+export class SnatchTag extends BattlerTag {
+  constructor() {
+    super(BattlerTagType.SNATCH, BattlerTagLapseType.TURN_END, 1);
```
Comment (Tempo-anon) [inline src/data/battler-tags/snatch-tag.ts]: Maybe `snatching` might be a better name since that clearly denotes that the battler tag owner is ready to snatch any moves. It provides just a big more info

**Review by Tempo-anon — DISMISSED** [review]
Tests pass. Weird minor edge case move

File: test/data/all-moves.test.ts [hunk 5]
```diff
@@ -43,12 +44,26 @@ describe("All Moves", async () => {
...
+   *   - Shelter
+   *   - Lunar Blessing
+   *   - Life Dew
+   *   - Clangorous Soul
+   *   - No Retreat
+   *   - Stuff Cheeks
+   *   - Take Heart
+   *   - Fillet Away
+   *   - Shed Tail
+   *   - Tidy Up
```
Comment (PigeonBar) [inline test/data/all-moves.test.ts:66]: Should Revival Blessing also be snatchable?

Comment (innerthunder) [reply test/data/all-moves.test.ts:66]: I guess so. I don't think it'd cause issues, at least

**Review by PigeonBar — APPROVED** [review]
